wolfcrypt/src/asn.c

```diff
@@ -1803,23 +1803,48 @@ int wc_CheckPrivateKey(byte* key, word32 keySz, DecodedCert* der)
 
     #ifdef HAVE_ECC
     if (der->keyOID == ECDSAk) {
-        word32  keyIdx = 0;
         ecc_key key_pair;
+        byte*   privDer;
+        word32  privSz;
+        word32  keyIdx = 0;
 
         if ((ret = wc_ecc_init(&key_pair)) < 0)
             return ret;
+
         if ((ret = wc_EccPrivateKeyDecode(key, &keyIdx, &key_pair,
                                                                  keySz)) == 0) {
             WOLFSSL_MSG("Checking ECC key pair");
-            keyIdx = 0;
-            if ((ret = wc_ecc_import_x963(der->publicKey, der->pubKeySize,
-                                                             &key_pair)) == 0) {
-                /* public and private extracted successfuly no check if is
+
+            if ((privSz = wc_ecc_size(&key_pair)) <= 0) {
+                return WC_KEY_SIZE_E;
+            }
+
+            privDer = (byte*)XMALLOC(privSz, der->heap, DYNAMIC_TYPE_KEY);
```

Comment:
The max ecc key size we have is 66 bytes (521 bits), why the overhead of dynamic memory.

Reply:
I agree, with knowing it is not going to be over 100 bytes I can use a buffer on the stack instead of malloc'ing. Will fix it with the MAX_ECC_BYTES macro.